libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -45,204 +45,303 @@ class AC_PosControl
-    ///     This function is private and contains all the shared z axis initialisation functions
+    // Fully initializes the U-axis controller with current position, velocity, acceleration, and attitude.
+    // Used during standard controller activation when full state is known.
+    // Private function shared by other vertical initializers.
     void init_U_controller();
 
-    /// input_accel_U_cm - calculate a jerk limited path from the current position, velocity and acceleration to an input acceleration.
-    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-    ///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_z.
+    // Sets the desired vertical acceleration in cm/s² using jerk-limited shaping.
+    // See input_accel_U_m() for full details.
     virtual void input_accel_U_cm(float accel_u_cmss);
 
-    /// input_vel_accel_U_cm - calculate a jerk limited path from the current position, velocity and acceleration to an input velocity and acceleration.
-    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-    ///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_z.
-    ///     The function modifies vel_u_cms to follow the jerk-limited trajectory defined by accel_u_cmss.
-    ///     The parameter limit_output specifies if the velocity and acceleration limits are applied to the sum of commanded and correction values or just correction.
+    // Sets the desired vertical acceleration in m/s² using jerk-limited shaping.
+    // Smoothly transitions to the target acceleration from current kinematic state.
+    // Constraints: max acceleration and jerk set via set_max_speed_accel_U_m().
+    virtual void input_accel_U_m(float accel_u_mss);
+
+    // Sets desired vertical velocity and acceleration (cm/s, cm/s²) using jerk-limited shaping.
+    // See input_vel_accel_U_m() for full details.
     virtual void input_vel_accel_U_cm(float &vel_u_cms, float accel_u_cmss, bool limit_output = true);
 
-    /// set_pos_target_U_from_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
-    ///     using the default jerk-limited kinematic shaping method.
-    ///     The zero target altitude is varied to follow pos_offset_u
+    // Sets desired vertical velocity and acceleration (m/s, m/s²) using jerk-limited shaping.
+    // Calculates required acceleration using current vertical kinematics.
+    // If `limit_output` is true, limits apply to the combined (desired + correction) command.
+    virtual void input_vel_accel_U_m(float &vel_u_ms, float accel_u_mss, bool limit_output = true);
```

Comment:
again not a blocker but I'm pretty sure we can remove the "virtual" from these two as well input_vel_accel_U_cm, input_vel_accel_U_m